### PR TITLE
fix: Optimize the custom-dropdown demo style

### LINKED ISSUE #39542: Dropdown custom render not working as expected
### Reproduction link

[https://ant.design/~demos/components-dropdown-demo-custom-dropdown](https://ant.design/~demos/components-dropdown-demo-custom-dropdown)

### Steps to reproduce

See documentation for Dropdown component and the "dropdownRender" function (https://ant.design/components/dropdown/#components-dropdown-demo-custom-dropdown). In the documentation page it looks ok, but if you open it in full screen: https://ant.design/~demos/components-dropdown-demo-custom-dropdown
you can see that the styling is off.

### What is expected?

I expected the Popover to include everything returned in the "dropdownRender" function. I expected there to be a white background out of the box.

### What is actually happening?

The "Click me" button in the example appears "outside" the popover.

| Environment | Info |
| --- | --- |
| antd | 5.0.7 |
| React | 18.2.0 |
| System | OSX |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by JarvisArt**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/39542

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Optimize the custom-dropdown demo style |
| 🇨🇳 Chinese |  优化dropdown自定义下拉demo样式  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/dropdown/demo/custom-dropdown.tsx [hunk 1]
```diff
@@ -32,26 +34,41 @@ const items: MenuProps['items'] = [
...
+
+  const menuStyle = {
+    boxShadow: 'none',
+  };
+
+  return (
+    <Dropdown
+      menu={{ items }}
+      dropdownRender={(menu) => (
+        <div style={contentStyle}>
```
Comment (afc163) [inline components/dropdown/demo/custom-dropdown.tsx:55]: 讲道理用了 dropdownRender 的话，这个阴影不应该让用户自己加。

Comment (JarvisArt) [reply components/dropdown/demo/custom-dropdown.tsx:55]: 之前做这个api是有在外层包一个div做这个阴影的，后面说会 break change，先不用加上，等反馈了再加上

Comment (JarvisArt) [reply components/dropdown/demo/custom-dropdown.tsx:55]: 想到了个办法，用户使用了dropdownRender 这个api时在给他外面包一div重置样式，这样就不会修改到常规dropdown的dom了。觉得这样可以的话我再改一下
image

File: components/dropdown/demo/custom-dropdown.tsx [hunk 2]
```diff
@@ -32,26 +34,41 @@ const items: MenuProps['items'] = [
...
-        <Space style={{ padding: 8 }}>
-          <Button type="primary">Click me!</Button>
+const App: React.FC = () => {
+  const { token } = useToken();
+
+  const contentStyle = {
+    backgroundColor: token.colorBgElevated, // You can use #fff
+    borderRadius: 8,
+    boxShadow:
+      '0 3px 6px -4px rgb(0 0 0 / 12%), 0 6px 16px 0 rgb(0 0 0 / 8%), 0 9px 28px 8px rgb(0 0 0 / 5%)',
```
Comment (MadCcc) [inline components/dropdown/demo/custom-dropdown.tsx:44]:
```suggestion
    borderRadius: token.borderRadiusLG,
    boxShadow: token.boxShadowSecondary,
```

File: components/dropdown/demo/custom-dropdown.tsx (hunk 1 above)
Comment (MadCcc) [reply components/dropdown/demo/custom-dropdown.tsx:55]: 还是先不加吧，这个 API 很久了，breaking change 不太好

**Review by MadCcc — APPROVED** [review]

File: components/dropdown/demo/custom-dropdown.tsx (hunk 1 above)
Comment (afc163) [reply components/dropdown/demo/custom-dropdown.tsx:54]: 不默认加导致我们自己也丢阴影了：https://github.com/ant-design/ant-design/issues/39799#issuecomment-1364820245

Comment (MadCcc) [reply components/dropdown/demo/custom-dropdown.tsx:54]: > 不默认加导致我们自己也丢阴影了：[#39799 (comment)](https://github.com/ant-design/ant-design/issues/39799#issuecomment-1364820245)

Table 这个阴影用错 token 了，应该是 `boxShadowSecondary`

**Pull request merged**